### ISSUE TITLE
MirBSD box

### ISSUE THREAD

**Issue by mcandre** [lavabit/robox#36]
Hi, I made a prototype base box for MirBSD, also known as MirOS. It's similar to OpenBSD but with a few quirks. Here are the packer tweaks and tests I implemented to support this VM:

https://github.com/mcandre/packer-templates/tree/master/miros

Basic functionality seems to work, including rsync-based shared folders and provisioning files and scripts. Some notes:

* ACPI support is missing. `packer build` is able to (just barely) succeed with the provided `boot_command`, but only because no complex applications are running at the same time that `halt -p` executes. The MirBSD devs are not interested in adding ACPI support, the VirtualBox developers are not interested in improving APM support. Applications will generally work, and file system can be generally safe, as long as sensitive user applications are safely terminated prior to `vagrant halt`. Not an ideal situation, but one we can document for the corner case of people wishing to deploy *MirBSD VM's* while avoiding accidental *file system corruption*! If you find that VMware or qemu provides better APM support, please let me know. Then it would actually make sense to offer VMware/libvirt providers but not VirtualBox, to safeguard against this issue.
* Related to ACPI lack of support, is the fact that much of the provisioning for the base box is done directly in `boot_command`, that would ordinarily be done at a later phase over SSH (much faster). This is because of how Packer doesn't understand that the last SSH provisioning script might trigger a final powerdown that is not recognized by ACPI. And so in practice Packer stalls the build waiting for an SSH connection that never happens. Fortunately, MirBSD is a fairly lean install (~100MiB) and so the `boot_command` provisioning is minimal in time and complexity.
* MirBSD's repository for install media including the base .NGZ files, is quite slow, taking around half an hour to download a basic set of install files. As a workaround, I have setup my packer build process with a `make` wrapper that mirrors these files locally with wget. It's up to you whether to do this; for debugging this really speeds up the build cycle. When everything is running smoothly, there's no particular reason to locally mirror these files.
* MirPorts appears to be broken. `pkg_add` still works, and I plan to look into pkgsrc later, in order to get more edgy packages like cmake and pip installed in some of my personal downstream boxes.

**Comment by ladar** [comment]
Looks interesting. I'm hoping to add 4 new build bots in the next couple of weeks, provided I can find RAM and SSDs for the recently donated blade server. I'm planning to add a few new box configs to the pipeline once that expansion is complete.
